test-dts/appUse.test-d.ts

```diff
@@ -0,0 +1,70 @@
+import { createApp } from './index'
+
+type App = ReturnType<typeof createApp>
+
+type PluginAOptionType = {
+  option1?: string
+  option2: number
+  option3: boolean
+}
+
+const PluginA = {
+  install(app: App, ...options: PluginAOptionType[]) {
```

Comment:
is there a reason you do this instead of 
```suggestion
  install(app: App, options: PluginAOptionType) {
```

Reply:
I was following the [original code](https://github.com/vuejs/core/pull/3969/files#diff-81c197a38d7dc6b7e764a1edffbdaf6bcb1885875e436d7c54e03a87731e2267L140), which declares `options` as a [rest parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters).

If I implemented your suggestion, it would require all plugins to implement `install()` with exactly one `options` argument. Currently, `install()` can be declared with 0, 1, or more `options` because it's a rest parameter.

Reply:
I've added a test below for passing multiple options since that's supported by the API.